### PR TITLE
Cleanup Parallel BoundaryCode

### PR DESCRIPTION
**Pull request by dschwoerer**
Use CRTP to avoid repeating the clone code.

It reduce the cxx part by ~ 40%

### CODE REVIEW

File: include/parallel_boundary_op.hxx [hunk 1]
```diff
@@ -3,9 +3,10 @@
 
 #include "boundary_op.hxx"
 #include "bout_types.hxx"
+#include "field_factory.hxx"
```
Comment (dschwoerer) [inline include/parallel_boundary_op.hxx:6]: Might be sufficient to forward declare FieldGenerator?

**Comment by ZedThree** [comment]
Neat, very clever! I need to find some time to look at this properly, but it looks good to me so far.

File: src/sys/generator_context.cxx [hunk 2]
```diff
@@ -45,5 +45,8 @@ Context::Context(const BoundaryRegion* bndry, int iz, CELL_LOC loc, BoutReal t,
   parameters["t"] = t;
 }
 
+Context::Context(BoutReal x, BoutReal y, BoutReal z, Mesh* msh, BoutReal t)
```
[the remark this answers is not in the document]
Comment (dschwoerer) [reply src/sys/generator_context.cxx:48]: I am not sure about renaming `t` - especially as we pass it as `{"t", t}`.
Should that be renamed to `time`?

**Comment by dschwoerer** [comment]
It has been a year. This is only 12 commits, of which 5 are merge commits, one is a merge fixup, and one is clang-format.

Thus 5 commits. @bendudson or @ZedThree, could one of you please have a look?

**Review by bendudson — APPROVED** [review]
Thanks @dschwoerer ! Sorry for the long delay.

**Pull request merged**